ProcessLib/DensityDrivenFlow/CreateDensityDrivenFlowProcess.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateDensityDrivenFlowProcess.h"
+
+#include "DensityDrivenFlowProcess.h"
+#include "DensityDrivenFlowProcessData.h"
+#include "ProcessLib/Parameter/ConstantParameter.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+namespace ProcessLib
+{
+namespace DensityDrivenFlow
+{
+std::unique_ptr<Process> createDensityDrivenFlowProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{process__type}
+    config.checkConfigParameter("type", "DensityDrivenFlow");
+
+    DBUG("Create DensityDrivenFlowProcess.");
+
+    // Process variable.
+    auto process_variables = findProcessVariables(
+        variables, config,
+        {"temperature_variable", "pressure_variable"});  // configure two Pcs
+
+    // Porosity parameter.
+    auto& porosity = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{process__DensityDrivenFlow__porosity}
+        "porosity", parameters, 1);
+    DBUG("Use \'%s\' as porosity parameter.", porosity.name.c_str());
+
+    // Parameter for the intrinsic permeability.
+    auto& intrinsic_permeability = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{process__DensityDrivenFlow__intrinsic_permeability}
+        "intrinsic_permeability", parameters, 1);
+    DBUG("Use \'%s\' as intrinsic_permeability parameter.",
+         intrinsic_permeability.name.c_str());
+
+    // Parameter for the storage.
+    auto& storage = findParameter<double>(
+        config,
+        //! \ogs_file_param_special{process__DensityDrivenFlow__storage}
+        "storage", parameters, 1);
+    DBUG("Use \'%s\' as storage parameter.", storage.name.c_str());
+
+    // Parameter for the reference_temperature.
+    auto& reference_temperature = findParameter<double>(
```

Comment:
minor: the name can be more specific, since it will be used in the fluid density model.